### PR TITLE
[SC-232] Add eth staking yield oracle

### PR DESCRIPTION
**Pull request by hexonaut**
Adds a staking yield oracle to be set by permissioned actor. We can discuss how this works in person.

### CODE REVIEW

**Comment by lucas-manuel** [comment]
At a glance this contract and the tests make sense, but not sure what this is actually for, why do we need the APR?

**Comment by hexonaut** [comment]
> At a glance this contract and the tests make sense, but not sure what this is actually for, why do we need the APR?

This is to be combined with the `RateTargetKinkInterestRateStrategy` to set the ETH market with a spread that is just below the staked eth yield. We can then set this value in one place (mainnet) and have the ETH market IRMs update across all networks without requiring a spell.

File: src/ETHStakingYieldOracle.sol [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.0;
+
+import { Ownable } from 'aave-v3-core/contracts/dependencies/openzeppelin/contracts/Ownable.sol';
+
+contract ETHStakingYieldOracle is Ownable {
```
Comment (lucas-manuel) [inline src/ETHStakingYieldOracle.sol:6]: Lets inherit and override IRateSource once the other PR is merged

File: test/ETHStakingYieldOracle.t.sol [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    address randomAddress = makeAddr("randomAddress");
+
+    ETHStakingYieldOracle oracle;
+
+    event BridgeUpdated(address indexed bridge, bool authorized);
+    event MaxAPRUpdated(uint256 maxAPR);
+    event APRUpdated(uint256 apr);
+
+    function setUp() public {
+        oracle = new ETHStakingYieldOracle();
```
Comment (lucas-manuel) [inline test/ETHStakingYieldOracle.t.sol:21]: transferOwnership to a makeAddr("owner"); address so permissioned calls have to be pranked and notOwner tests don't prank

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

File: test/ETHStakingYieldOracle.t.sol [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    function test_onReceiveData_not_bridge() public {
+        oracle.setMaxAPR(1e27);
+        oracle.setAuthorizedBridge(bridge, true);
+
+        vm.prank(randomAddress);
+        vm.expectRevert("ETHStakingYieldOracle/not-authorized");
+        oracle.onReceiveData(abi.encode(uint256(0.03e27)));
+    }
+
+    function test_onReceiveData_invalid_apr() public {
```
Comment (lucas-manuel) [inline test/ETHStakingYieldOracle.t.sol:85]: Tests look good, lets just reorder to put failure modes above success cases and in order of execution

E.g., onReceiveData

test_onReceiveData_not_bridge
test_onReceiveData_invalid_apr
test_onReceiveData

File: test/ETHStakingYieldOracle.t.sol (hunk 2 above)
Comment (hexonaut) [reply test/ETHStakingYieldOracle.t.sol:21]: fixed

File: test/ETHStakingYieldOracle.t.sol (hunk 3 above)
Comment (hexonaut) [reply test/ETHStakingYieldOracle.t.sol:85]: fixed

**Comment by hexonaut** [comment]
This shouldn't have been marked for review actually. I want to use the `AccessControl` for the auth as well as add the IRateSource interface as requested. Will address all this after the other PR is merged.

File: src/ETHStakingYieldOracle.sol [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        _apr = apr;
+
+        emit APRUpdated(apr);
+    }
+
+    function getAPR() external view returns (uint256) {
+        return _apr;
+    }
+
+}
```
Comment (barrutko) [inline src/ETHStakingYieldOracle.sol:46]: Missing empty line at the end

File: src/ETHStakingYieldOracle.sol (hunk 1 above)
Comment (barrutko) [reply src/ETHStakingYieldOracle.sol:6]: I'd actually inherit proper interfaces even in this PR. It'll make it clear why `_apr` is internal and then we have public `getAPR `, why there is `onReceiveData` etc. At this point it's a little unclear, where everything comes from. It's a little more complex here than just `latestAnswer` in oracle conract.

**Review by barrutko — COMMENTED** [review]
Generally makes sense. Commented on some minor stuff.

**Comment by hexonaut** [comment]
This is on hold. Don't worry about reviewing.

**Comment by hexonaut** [comment]
Actually, this is no longer necessary. Closing.

**Pull request closed without merging**